lib/portlayer/exec/container.go

```diff
@@ -563,7 +616,9 @@ func (c *Container) LogReader(ctx context.Context, tail int, follow bool) (io.Re
 		}
 	}
 
-	if follow && c.State == StateRunning {
+	c.m.Lock()
```

Comment:
Makes sense. However, I am paying attention to the managing a state. Nevertheless, I addressed it.